src/olympia/devhub/forms.py

```diff
@@ -557,23 +540,31 @@ class StandaloneValidationForm(AddonUploadForm):
             u'your own and only need it to be signed by Mozilla.'))
 
 
-class NewVersionForm(NewAddonForm):
+class NewVersionForm(AddonUploadForm):
+    supported_platforms = forms.TypedMultipleChoiceField(
+        choices=amo.SUPPORTED_PLATFORMS_CHOICES,
+        widget=forms.CheckboxSelectMultiple(attrs={'class': 'platform'}),
+        initial=[amo.PLATFORM_ALL.id],
+        coerce=int,
+        error_messages={'required': 'Need at least one platform.'}
+    )
+
     beta = forms.BooleanField(
         required=False,
         help_text=_lazy(u'A file with a version ending with '
                         u'a|alpha|b|beta|pre|rc and an optional number is '
                         u'detected as beta.'))
 
     def __init__(self, *args, **kw):
-        self.addon = kw.pop('addon')
+        self.addon = kw.pop('addon', None)
```

Comment:
Ah... in `_submit_upload` the `addon` parameter is `None` for new Addon uploads, and that's passed onto `forms.NewVersionForm`.  So I can either change that to not pass the `addon` parameter if `None` or take out the `self.addon = kw.pop('addon', None)` in the form `__init__`